src/lib/index.js

```diff
@@ -175,7 +176,33 @@ function maybeAdjustSrc(src, srcAttributePath, traceType, config) {
   return config && config.fromSrc ? config.fromSrc(src, traceType) : src;
 }
 
+function adjustColorscale(
+  colorscale,
+  numberOfNeededColors,
+  colorscaleType,
+  config
+) {
+  if (config.repeat) {
+    const repetitions = Math.ceil(numberOfNeededColors / colorscale.length);
+    const newArray = new Array(repetitions).fill(colorscale);
+    return newArray
+      .reduce((a, b) => {
+        return a.concat(b);
+      }, [])
+      .slice(0, numberOfNeededColors);
```

Comment:
What do you need that newArray for?

```
    return new Array(repetitions).fill(colorscale
      .reduce((a, b) => a.concat(b), [])
      .slice(0, numberOfNeededColors);
```

Reply:
its just my way of generating an array of colors that repeat themselves enough times to cover all the traces that need to be colored..

Reply:
yeah, I figured, just saying could be written sorter without extra consts